src/winmain.cpp

```diff
@@ -917,6 +917,43 @@ bool runInstaller(const wstring& app2runPath, const wstring& binWindowsClassName
 	return true;
 }
 
+// Returns a folder suitable for exe installer download destination.
+std::wstring getDestDir()
+{
+	// Some users might unset the temp variable, so need to account for that.
+	// Other fallbacks: %AppData%, %UserProfile% or GetModuleFileName().
+	size_t requiredSize = 0;
+	std::vector<wchar_t> envPath(MAX_PATH + 1, 0);
+	for (const wchar_t* env : { L"TEMP", L"TMP" })
+	{
+		errno_t e = _wgetenv_s(&requiredSize, envPath.data(), envPath.size(), env);

```

Comment:
`_wgetenv_s` is a more secure alternative to `_wgetenv`.
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getenv-s-wgetenv-s?view=msvc-160
> These versions of getenv, _wgetenv have security enhancements, as described in Security Features in the CRT.